tests/test_scalars.py

```diff
@@ -51,6 +51,339 @@
+    check(assert_type(1.5 + interval_f, "pd.Interval[float]"), pd.Interval)
+    check(
+        assert_type(interval_ts + pd.Timedelta(days=1), "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(pd.Timedelta(days=1) + interval_ts, "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(interval_td + pd.Timedelta(days=1), "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(pd.Timedelta(days=1) + interval_td, "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+    )
+
+
+def test_interval_cmp():
+    interval_i = pd.Interval(0, 1, closed="left")
+    interval_f = pd.Interval(0.0, 1.0, closed="right")
+    interval_ts = pd.Interval(
+        pd.Timestamp("2017-01-01"), pd.Timestamp("2017-01-02"), closed="both"
+    )
+    interval_td = pd.Interval(
+        pd.Timedelta("1 days"), pd.Timedelta("2 days"), closed="neither"
+    )
+
+    check(assert_type(0.5 in interval_i, bool), bool)
+    check(assert_type(1 in interval_i, bool), bool)
+    check(assert_type(1 in interval_f, bool), bool)
+    check(assert_type(pd.Timestamp("2000-1-1") in interval_ts, bool), bool)
+    check(assert_type(pd.Timedelta(days=1) in interval_td, bool), bool)
+
+    interval_index_int = pd.IntervalIndex([interval_i])
+    check(
+        assert_type(interval_index_int >= interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_index_int < interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_index_int <= interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_index_int > interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+
+    check(
+        assert_type(interval_i >= interval_index_int, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_i < interval_index_int, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_i <= interval_index_int, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_i > interval_index_int, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+
+    check(
+        assert_type(interval_index_int == interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(interval_index_int != interval_i, np_ndarray_bool),
+        np.ndarray,
+        np.bool_,
+    )
+
+    check(
+        assert_type(
+            interval_i == interval_index_int,
+            np_ndarray_bool,
+        ),
+        np.ndarray,
+        np.bool_,
+    )
+    check(
+        assert_type(
+            interval_i != interval_index_int,
+            np_ndarray_bool,
+        ),
+        np.ndarray,
+        np.bool_,
+    )
+
```

Comment:
It would be nice to add tests that check that the typing is not too wide.  Don't have to do it for this PR.  E.g., `interval_i >= pd.Timestamp("2017-01-01")` should be disallowed.